OIPS/OIP-0010.md

```diff
@@ -0,0 +1,117 @@
+Instead of mosaic's set of staked PoS validators, the anchor relies solely on an
+organization to provide state roots across chains.
+
+Depending on the use-case, an anchor or a mosaic core may be preferred. An
+anchor makes it much, much easier to deploy a side chain as compared to a mosaic
+core.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations.-->
+An `Anchor` provides a method to anchor a state root. It also implements the
+`StateRootInterface` to make the state root later available on-chain.
+
+```solidity
+/**
+ *  @notice Anchor new state root for a block height.
+ *
+ *  @param _blockHeight Block height at which the state root was valid.
+ *  @param _stateRoot State root at given block height.
+ *
+ *  @return succes_ Whether anchoring was successful.
+ */
+function anchorStateRoot(
+    uint256 _blockHeight,
+    bytes32 _stateRoot
+)
+    external
+    onlyOrganization
+    returns (bool success_);
+```
+
+```solidity
+interface StateRootInterface {
+
+    /**
+     * @notice Gets the block number of latest committed state root.
+     *
+     * @return height_ Block height of the latest committed state root.
+     */
+    function getLatestStateRootBlockHeight()
+        external
+        view
+        returns (uint256 height_);
+
+    /**
+     * @notice Get the state root for the given block height.
+     *
+     * @param _blockHeight The block height for which the state root is fetched.
+     *
+     * @return bytes32 State root at the given height.
+     */
+    function getStateRoot(uint256 _blockHeight)
+        external
+        view
+        returns (bytes32 stateRoot_);
+
+}
+```
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+The method `anchorStateRoot` is kept simple on purpose. The state root and the
+height of the block that contains that state root are sufficient to prove a
+certain state at a certain height. A relevant state to prove could be the fact
+that value was locked into a gateway or co-gateway.
+
+We decided to implement the `StateRootInterface` with the Anchor as this enables
+interchangeability between implementations that provide the state root to other
+contracts, for example the gateway or co-gateway.
+
+An alternative to the anchor is a mosaic core. The main difference is that the
+mosaic core requires a set of staked validators. The anchor on the other hand
+requires only an "organization" to anchor a state root.
+
+## Implementation
```

Comment:
Not sure, but do we need a section here for a circular buffer of state roots. So that it doesn't infinitely append state roots?

Reply:
Good point 👍 We should mention that our current implementation does that.